kubeshop/testkube-dashboard
# tech: split entity details components

This PR is meant to untangle most of the test and test suite details. Previously, all the components there were based on an abstract configuration, and the logic that had to be split between different types was just using conditionals on `entity` (`'tests' | 'test-suites'`).

This step decomposes tests and test suites, so these may be composed later better, even sharing the components and structure with other types (test sources, executors and triggers).

Unfortunately, splitting that wasn't simple - many disentanglements were leading to refactoring of some internal logic, to make it simpler, and working well without previous conditionals.

## Changes

- Split `organisms/EntityDetails/*` components into `pages/Tests/*` and `pages/TestSuites/*`
  - There are still some leftovers - there were places where the logic was actually the same, and it wasn't worth duplicating it fully yet *(the PR is already big enough)*
  - We may get to a point where it will be possible, but this will happen most likely after improving composition of forms
- Unify common settings sections
  - Both in case of Tests / Test Suites, and Executors / Test Sources / Triggers
- Use Test Suites converted to V3 already in Zustand
  - Thanks to that UI doesn't have to care about it that much
- Delete `@constants/entityDetailsConfig` along with `useEntityDetailsConfig`
  - Now components either are entity-specific, or have the implementation provided by props (i.e. `useAbortExecution`)
  - Thanks to that, these will have more in common with Test Sources / Triggers / Executors

### Misc changes

- Add `useDashboardNavigate` hook
  - There was commonly duplicated logic of getting `navigate` from `DashboardContext`, building a callback with `useCallback`/`useLastCallback` and passing it down
- Flatten loading pages
  - These were anyway barely lazy loading, yet it didn't look well
  - We may improve it after https://github.com/kubeshop/testkube/issues/4249
- Simplify the logic for prettifying arguments
- Simplify the logic (basically duplicated) for displaying execution time
- Delete duplicated `<Space>` from all settings - use `gap` on top-level instead

## Fixes

- https://github.com/kubeshop/testkube/issues/3958

## How to test it

- Go to the [**Vercel preview**](https://testkube-dashboard-git-dawid-techsplit-en-22e041-testkube-cloud.vercel.app/tests?~api_server_endpoint=https://dev.testkube.io/results/v1) and see how it works

## Checklist
